mempool/src/feerate.rs

```diff
@@ -0,0 +1,72 @@
+use std::num::NonZeroU128;
+
+use common::primitives::amount::Amount;
+
+lazy_static::lazy_static! {
+    pub(crate) static ref INCREMENTAL_RELAY_FEE_RATE: FeeRate = FeeRate::new(Amount::from_atoms(1000));
+    pub(crate) static ref INCREMENTAL_RELAY_THRESHOLD: FeeRate = FeeRate::new(Amount::from_atoms(500));
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct FeeRate {
+    atoms_per_kb: u128,
+}
+
+impl FeeRate {
+    pub(crate) fn new(tokens_per_kb: Amount) -> Self {
+        Self {
+            atoms_per_kb: tokens_per_kb.into_atoms(),
+        }
+    }
+
+    pub(crate) fn of_tx(fee: Amount, tx_size: usize) -> Self {
+        Self {
+            atoms_per_kb: Self::div_up(1000 * fee.into_atoms(), tx_size),
+        }
+    }
+
+    pub(crate) fn compute_fee(&self, size: usize) -> Amount {
+        let size = u128::try_from(size).expect("compute_fee conversion");
+        Amount::from_atoms(self.atoms_per_kb * size / 1000)
```

Comment:
Two things:
- Division has higher priority than multiplication, which will make (size/1000) happen first, which means that the outcome will become zero in if size < 1000, which I'm sure isn't the outcome we want. Unlikely, but possible I'd say. At best it's optimistic. So this has to be changed to `((self.atoms_per_kb * size) / 1000)`.
- In bitcoin, there's a ceil operation being done on this. The ceil can be done here for integers with `((self.atoms_per_kb * size + 999) / 1000)`. Perhaps write a comment that 999 is for ceil calculation, or wrap this somehow with a function that explains it.

Reply:
Done.